### PR TITLE
Rvl 1431

### LINKED ISSUE #1242: END FUND Kenya - Create CDD Supervisor Daily Summary form
Jira ticket: https://smartregister.atlassian.net/browse/RVL-1431 

The apk should have the ability to select locations at levels higher than the structure level e.g. at ward level

- This form should be accessed by tapping on the ward boundary ( something similar to how the IRS lite works).
- A user should be able to enter multiple form instances by tapping on the same ward. So the user should be able to tap on the ward and capture data for a CDD for a specific date and save, then tab on the same ward again to enter data for the same CDD for a different date or for another CDD for a specific date.

The form fields and logic can be accessed [here](https://docs.google.com/spreadsheets/d/1J5IsHWSznKEdT82bxGY6Yzfwi43su34mkX0w8bkAeRU/edit#gid=1800425965).

- [x] Create form
- [x] Wire it up to be accessible when a ward is clicked

### PR DESCRIPTION
**Pull request by lindagcaba**
resolves #1242

### CODE REVIEW

**Comment by lindagcaba** [comment]
also resolves #1243

File: opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java [hunk 1]
```diff
@@ -351,7 +353,10 @@ private void onFeatureSelectedByNormalClick(Feature feature) {
             listTaskInteractor.fetchFamilyDetails(selectedFeature.id());
         } else if (IRS_VERIFICATION.equals(code) && isZambiaIRSLite()) {
             listTaskInteractor.fetchInterventionDetails(IRS, feature.id(), false);
-        } else if (IRS_VERIFICATION.equals(code) && COMPLETE.equals(businessStatus)) {
+        }else if (CDD_SUPERVISION.equals(code) && isKenyaMDALite() && (NOT_VISITED.equals(businessStatus) || IN_PROGRESS.equals(businessStatus))){
+            startForm(selectedFeature, null, selectedFeatureInterventionType);
```
Comment (Rkareko) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java:357]: Could we use `listTaskInteractor.fetchInterventionDetails` so that validation of user location is done when `validateFarStructures` is set to true

File: opensrp-reveal/src/test/java/org/smartregister/reveal/util/RevealJsonFormUtilsTest.java [hunk 2]
```diff
@@ -510,7 +513,7 @@ public void testPopulateVerificationFormWithServerOptions() throws JSONException
         verify(revealJsonFormUtils).populateServerOptions(serverConfigs, Constants.CONFIGURATION.FIELD_OFFICERS, fieldsMap.get(JsonForm.FIELD_OFFICER), "Lusaka");
     }
 
-    @Test
+
```
Comment (Rkareko) [inline opensrp-reveal/src/test/java/org/smartregister/reveal/util/RevealJsonFormUtilsTest.java:516]: Why was the test annotation removed?

Comment (lindagcaba) [reply opensrp-reveal/src/test/java/org/smartregister/reveal/util/RevealJsonFormUtilsTest.java:516]: merge, error this one. fixing.

File: opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java (hunk 1 above)
Comment (lindagcaba) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java:357]: updating this...

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java [hunk 3]
```diff
@@ -133,7 +135,7 @@ public void fetchInterventionDetails(String interventionType, String featureId,
         } else if (IRS_VERIFICATION.equals(interventionType)) {
             sql = String.format("SELECT %s, %s, %s, %s, %s, %s FROM %s WHERE id= ?",
                     TRUE_STRUCTURE, ELIGIBLE_STRUCTURE, REPORT_SPRAY, CHALK_SPRAY, STICKER_SPRAY, CARD_SPRAY, IRS_VERIFICATION_TABLE);
-        } else if (REGISTER_FAMILY.equals(interventionType)) {
+        } else if (REGISTER_FAMILY.equals(interventionType) || CDD_SUPERVISION.equals(interventionType)) {
```
Comment (Rkareko) [inline opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java:138]: Is this change required?

Comment (lindagcaba) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java:138]: yes no longer required.will remove

File: opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java (hunk 1 above)
Comment (lindagcaba) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/ListTaskPresenter.java]: updated

File: opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java (hunk 3 above)
Comment (lindagcaba) [reply opensrp-reveal/src/main/java/org/smartregister/reveal/interactor/ListTaskInteractor.java]: removed this

**Review by Rkareko — APPROVED** [review]
